management-api-server/src/main/java/com/datastax/mgmtapi/resources/v1/TableOpsResources.java

```diff
@@ -312,4 +312,105 @@ public Response list(
           return Response.ok(tables, MediaType.APPLICATION_JSON).build();
         });
   }
+
+  @POST
+  @Path("/garbagecollect")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.TEXT_PLAIN)
+  @ApiResponse(
+      responseCode = "202",
+      description = "Job ID for table garbage collection process",
+      content =
+          @Content(
+              mediaType = MediaType.TEXT_PLAIN,
+              schema = @Schema(implementation = String.class),
+              examples = @ExampleObject(value = "d69d1d95-9348-4460-95d2-ae342870fade")))
+  @ApiResponse(
+      responseCode = "400",
+      description = "Invalid table garbage collection request",
+      content =
+          @Content(
+              mediaType = MediaType.TEXT_PLAIN,
+              schema = @Schema(implementation = String.class),
+              examples = @ExampleObject(value = "tombstoneOption must be either ROW or CELL")))
+  @Operation(
+      summary = "Remove deleted data from one or more tables",
+      operationId = "garbageCollect")
+  public Response garbageCollect(
+      @QueryParam(value = "tombstoneOption") String tombstoneOptionStr,
+      KeyspaceRequest keyspaceRequest) {
+    return handle(
+        () -> {
+          List<String> tables = keyspaceRequest.tables;
+          if (CollectionUtils.isEmpty(tables)) {
+            tables = new ArrayList<>();
+          }
+
+          String keyspaceName = keyspaceRequest.keyspaceName;
+          if (StringUtils.isBlank(keyspaceName)) {
+            keyspaceName = "ALL";
+          }
```

Comment:
issue:
If the request specifies a keyspace that does not exist, the Cassandra Storage service will throw an exception, which the NodeOpsProvider in the Agent will catch and throw a RuntimeException, which will bubble up in this `handle` call and result in one of our favorite `500 Internal Server Errors`. 

Could you add a check here and maybe return either a 404 or a 400 if the keyspace in the request does not exist? You can use the common method here to test if it exists:
https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/common/BaseResources.java#L85-L96
An example of how we use it is here:
https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/v2/TokenResourcesV2.java#L60-L62
Note that you will have to update the annotations if you send back a 404. I'm fine with a 400 in this case, so long as the message indicate the keyspace doesn't exist.